internal/agent/httpserver/negotiator.go

```diff
@@ -0,0 +1,176 @@
+// This file was copied from go-ozzo/ozzo-routing
+// The goal is to do the Accept header parsing for the ContentType negotiation.
+
+package httpserver
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+// AcceptRange represents an accept range as defined in https://tools.ietf.org/html/rfc7231#section-5.3.2
+//
+// Accept = #( media-range [ accept-params ] )
+//  media-range    = ( "*/*"
+//                   / ( type "/" "*" )
+//                   / ( type "/" subtype )
+//                   ) *( OWS ";" OWS parameter )
+//  accept-params  = weight *( accept-ext )
+//  accept-ext = OWS ";" OWS token [ "=" ( token / quoted-string ) ]
+type AcceptRange struct {
+	Type       string
+	Subtype    string
+	Weight     float64
+	Parameters map[string]string
+	raw        string // the raw string for this accept
+}
+
+// RawString returns the raw string in the request specifying the accept range.
+func (a AcceptRange) RawString() string {
+	return a.raw
+}
+
+// AcceptMediaTypes builds a list of AcceptRange from the given HTTP request.
+func AcceptMediaTypes(r *http.Request) []AcceptRange {
```

Comment:
This source file is copied from go-ozzo/ozzo-routing, but I had to change it since we need to return some default media type and this case wasn't supported. I've copied the tests from the same location here as well.